### PR TITLE
Fix/casi live views

### PR DESCRIPTION
**Pull request by guoen21**
Add two player views for live csai playback.

### CODE REVIEW

File: csai/src/main/java/com/dice/doris/csai/MainActivity.kt [hunk 1]
```diff
@@ -39,44 +40,44 @@ class MainActivity : AppCompatActivity(), SourceCallback {
...
             ExoDorisImaCsaiBuilder(this@MainActivity).apply {
-                setAdViewProvider(playerView)
+                setAdViewProvider(adViewProvider)
             }
-        } else if (adType === DorisAdEvent.AdType.IMA_CSAI_LIVE) {
+        } else if (adType === AdType.IMA_CSAI_LIVE) {
             ExoDorisImaCsaiLiveBuilder(this@MainActivity).apply {
-                setLiveAdSurfaceView(playerView.videoSurfaceView as SurfaceView)
-                setAdViewProvider(playerView)
+                setAdViewProvider(adViewProvider)
```
Comment (hongfeih-es) [inline csai/src/main/java/com/dice/doris/csai/MainActivity.kt]: seems `adViewProvider` is not needed.

```suggestion
                setAdViewProvider(secondaryPlayerView)
```

File: csai/src/main/java/com/dice/doris/csai/MainActivity.kt [hunk 2]
```diff
@@ -39,44 +40,44 @@ class MainActivity : AppCompatActivity(), SourceCallback {
...
             .setDrmParams(videoInfo.drm?.let { drm -> ActionToken(CsaiConfig.videoId, videoInfo.url, drm.url, drm.jwtToken, "widevine") })
             .setImaCsaiProperties(imaCsaiProperties)
             .setTextTracks(parseTextTrack(videoInfo))
             .build()
 
-        player = createPlayer(Source.getAdType(src))
+        val adType = Source.getAdType(source)
+        // Currently we use two player views for csai live playback because can not get SurfaceView.
+        // When our doris ui supports TV device and then we should not use two views.
+        val adViewProvider: AdViewProvider =
```
Comment (hongfeih-es) [inline csai/src/main/java/com/dice/doris/csai/MainActivity.kt]: adViewProvider can be get by adType, so seems not needed.

File: csai/src/main/java/com/dice/doris/csai/MainActivity.kt [hunk 3]
```diff
@@ -107,6 +108,45 @@ class MainActivity : AppCompatActivity(), SourceCallback {
...
+                    val adMarkers = adEvent.details.adMarkers
+                    playerView.setExtraAdGroupMarkers(
+                        adMarkers.adGroupTimesMs,
+                        adMarkers.playedAdGroups
+                    )
+                }
+
+            else -> {}
+        }
+    }
```
Comment (hongfeih-es) [inline csai/src/main/java/com/dice/doris/csai/MainActivity.kt:143]:
```suggestion
    }

```

**Review by hongfeih-es — APPROVED** [review]

File: csai/README.md [hunk 4]
```diff
@@ -33,11 +33,17 @@ const val CMP_USP = ""  // TODO: put CCPA us privacy string from Consent Provide
...
 
-#### 1. Get your auth token first.
+#### Get your auth token first
+Need to call login to get your auth token for following steps.
 
-#### 2. Get video detail (sample code use okhttp client.)
+#### Get video detail (sample code use okhttp client.)
 
-- Client macros needed for customized ads:
+Your auto token is required to get video details.
```
Comment (szaboa) [inline csai/README.md]: typo - auth and not auto

**Review by szaboa — APPROVED** [review]

File: csai/README.md (hunk 4 above)
Comment (guoen21) [reply csai/README.md]: done

**Pull request merged**